src/cargo/core/workspace.rs

```diff
@@ -201,6 +201,17 @@ impl<'cfg> Workspace<'cfg> {
         }.parent().unwrap()
     }
 
+    /// Returns the root path of this workspace.
+    ///
+    /// That is, this returns the path of the directory containing the
+    /// `Cargo.toml` which is the root of this workspace.
+    pub fn workspace_dir(&self) -> Option<&Path> {
+        match self.root_manifest {
```

Comment:
I think this should always be available, right? In that this function should return `&Path` instead of `Option<&Path>`? (sort of how `root` above cannot fail)